Dockerfile

```diff
@@ -9,25 +9,22 @@ ENV MUESLI_PATH=/opt/muesli4
 EXPOSE 8080
 CMD ["/opt/muesli4/docker-serve.sh"]
 
-RUN useradd muesli
+RUN useradd muesli && \
+    apt-get update && \
+    apt-get install -y \
+        python3.6 python3.6-dev lp-solve postgresql-server-dev-10 wget  \
+        python3-pip libjs-prototype libjs-select2.js libjs-jquery-fancybox \
+        libjs-jquery-tablesorter locales libpcre3 libpcre3-dev && \
+    rm -rf /var/lib/apt/lists/* && \
+    locale-gen de_DE.UTF-8 && \
+    wget https://www.mathi.uni-heidelberg.de/~jvisintini/lp_solve -O /usr/bin/lp_solve && \
+    wget https://www.mathi.uni-heidelberg.de/~jvisintini/libxli_DIMACS.so -O /usr/lib/lp_solve/libxli_DIMACS.so
 
-RUN apt-get update && \
-apt-get install -y python3.6 python3.6-dev lp-solve \
-postgresql-server-dev-10 wget python3-pip libjs-prototype \
-libjs-select2.js libjs-jquery-fancybox libjs-jquery-tablesorter \
-locales libpcre3 libpcre3-dev && \
-rm -rf /var/lib/apt/lists/*
-
-RUN locale-gen de_DE.UTF-8
 ENV LANG de_DE.UTF-8
 ENV LANGUAGE de_DE:de
 ENV LC_ALL de_DE.UTF-8
 
-RUN wget https://www.mathi.uni-heidelberg.de/~jvisintini/lp_solve -O /usr/bin/lp_solve
-RUN wget https://www.mathi.uni-heidelberg.de/~jvisintini/libxli_DIMACS.so -O /usr/lib/lp_solve/libxli_DIMACS.so
-
-
-RUN pip3 install --upgrade pip
 COPY --chown=muesli:muesli ./requirements.txt /opt/muesli4/
-RUN pip3 install -r requirements.txt
+RUN python3 -m pip install --upgrade pip && \
```

Comment:
Well yeah, If you think about the developement workflow this would probably better seperated, but in the docker documentation it said that it is best practise to use a minimal amount of layers: https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#minimize-the-number-of-layers

After reading this I grouped all the commands that create layers :blush:

Reply:
Makes sense. Let's follow the recommended way